dspace-api/src/main/java/org/dspace/util/SolrImportExport.java

```diff
@@ -0,0 +1,633 @@
+		}
+
+		String timeField = makeTimeField(indexName);
+
+		// Create a temp directory to store temporary core data
+		File tempDataDir = new File(ConfigurationManager.getProperty("dspace.dir") + File.separator + "temp" + File.separator + "solr-data");
+		boolean tempDirCreated = tempDataDir.mkdirs();
+		if (!tempDirCreated)
+		{
+			throw new SolrImportExportException("Could not create temporary data directory " + tempDataDir.getCanonicalPath());
+		}
+
+		try
+		{
+			// create a temporary core to hold documents coming in during the reindex
+			HttpSolrServer adminSolr = new HttpSolrServer(baseSolrUrl);
+			CoreAdminRequest.Create createRequest = new CoreAdminRequest.Create();
+			createRequest.setInstanceDir(solrInstanceDir);
+			createRequest.setDataDir(tempDataDir.getCanonicalPath());
+			createRequest.setCoreName(tempIndexName);
+
+			int responseCode = createRequest.process(adminSolr).getStatus();
+			if (responseCode != 0)
+			{
+				// the "process" call has probably thrown an exception already -- throw this just in case
+				throw new SolrImportExportException("Could not create temporary core on Solr server " + baseSolrUrl + ", code was: " + responseCode);
+			}
+
+			// swap actual core with temporary one
+			CoreAdminRequest swapRequest = new CoreAdminRequest();
+			swapRequest.setCoreName(indexName);
+			swapRequest.setOtherCoreName(tempIndexName);
+			swapRequest.setAction(CoreAdminParams.CoreAdminAction.SWAP);
+			swapRequest.process(adminSolr);
+
+			// export from the actual core (from temp core name, actual data dir)
+			exportIndex(indexName, exportDir, tempSolrUrl, timeField);
+
+			// clear actual core (temp core name, clearing actual data dir) & import
+			importIndex(indexName, exportDir, tempSolrUrl, true, true);
+
+			// commit changes
+			HttpSolrServer origSolr = new HttpSolrServer(origSolrUrl);
+			origSolr.commit();
+
+			// swap back (statistics now going to actual core name in actual data dir)
+			swapRequest = new CoreAdminRequest();
+			swapRequest.setCoreName(tempIndexName);
+			swapRequest.setOtherCoreName(indexName);
+			swapRequest.setAction(CoreAdminParams.CoreAdminAction.SWAP);
+			swapRequest.process(adminSolr);
+
+			// export all docs from now-temp core into a temporary directory...
+			String tempExportDirName = ConfigurationManager.getProperty("dspace.dir") + File.separator + "temp" + File.separator + "solr-data-tmp";
+			exportIndex(tempIndexName, tempExportDirName, tempSolrUrl, timeField);
+			// ...and import them into the now-again-actual core *without* clearing
+			importIndex(tempIndexName, tempExportDirName, origSolrUrl, false, true);
+
+			// commit changes
+			origSolr.commit();
+
+			// delete temporary export dir
+			FileUtils.deleteDirectory(new File(tempExportDirName));
+
+			// unload now-temp core (temp core name)
+			CoreAdminRequest.unloadCore(tempIndexName, false, false, adminSolr);
+
+			if (!keepExport)
+			{
+				FileUtils.deleteDirectory(new File(exportDir));
```

Comment:
Thanks @mwoodiupui  / @hardyoyo , yes that would definitely be cleaner. I'm not sure I'll have time today to change the code, so feel free to push to the PR branch if you have time.